AmericanMedicalAssociation/ama-style-guide-2
# Exclude pattern scaffolding SCSS from d8

## Ticket(s)

N/A (quickish fix while working on [EWL-4166](https://issues.ama-assn.org/browse/EWL-4166))

## Description

`pattern-scaffolding.scss` should not be included in the d8 assets. We only want the styles there to affect PatternLab, not show up in the Drupal theme.

- Changes the glob import rule in `styles.scss` to only include files that start with an `_` (This should include all pattern files, but not include those that don't look like `_foo.scss`. 
- This will specifically not include `pattern-scaffolding.scss` since that file doesn't start with an underscore - however, the file will still be compiled to CSS and rendered in PatternLab in the browser.

## To Test

- [ ] Follow the testing steps in https://github.com/AmericanMedicalAssociation/ama-d8/pull/195 to connect your local SG2/D8 environments for testing (so you can see the styles from this SG2 PR in d8 - preferably, use develop for the d8 repo)
- [ ] open any SCSS file starting with an underscore (e.g. 01-atoms/_whatever.scss) and make an obvious styling change, then run `gulp`:
- - [ ] open http://localhost:3000/ in a browser and see that your change is visible
- - [ ] open http://ama-d8.local/ and see that your change is visible
- [ ] open `pattern-scaffolding.scss` and make an obvious styling change, then run `gulp`:
- - [ ] open http://localhost:3000/ in a browser and see that your change is visible
- - [ ] open http://ama-d8.local/ and see that your change is NOT visible in d8

## Visual Regressions

N/A. Visual regression testing in SG2/D8 is dependent upon https://issues.ama-assn.org/browse/EWL-3503.


## Relevant Screenshots/GIFs

N/A


## Remaining Tasks

N/A


## Additional Notes

Anything more to add? Good things to call out here are:
- this blocks https://github.com/AmericanMedicalAssociation/ama-style-guide-2/pull/29
